contents/tutorials/redirect-testing.md

```diff
@@ -0,0 +1,398 @@
+
+```js
+// app/providers.js
+'use client'
+import posthog from 'posthog-js'
+import { PostHogProvider } from 'posthog-js/react'
+
+if (typeof window !== 'undefined') {
+  posthog.init("<ph_project_api_key>", {
+    api_host: "<ph_instance_address>"
+  })
+}
+
+export default function PHProvider({ children }) {
+  return <PostHogProvider client={posthog}>{children}</PostHogProvider>
+}
+```
+
+Import the `PHProvider` component into `layout.js` and wrap your app in it.
+
+```js
+import './globals.css'
+import PHProvider from './providers'
+
+export default function RootLayout({ children }) {
+  return (
+    <html lang="en">
+      <PHProvider>
+        <body>{children}</body>
+      </PHProvider>
+    </html>
+  )
+}
+```
+
+Once set up, PostHog autocaptures usage and you can use all its tools throughout your app.
+
+## Adding test pages
+
+In the `app` folder, create two new folders named `control` and `test`. In each of them, create a basic `page.js` file with a button to capture an event. This is what the control page looks like:
+
+```js
+// app/control/page.js
+'use client'
+import { usePostHog } from "posthog-js/react";
+
+export default function Control() {
+  const posthog = usePostHog();
+
+  return (
+    <main>
+      <h1>Hello!</h1>
+      <button onClick={() => posthog.capture("main_button_clicked")}>
+        Click me!
+      </button>
+    </main>
+  );
+}
+```
+
+This is what the test page looks like:
+
+```js
+// app/test/page.js
+'use client'
+import { usePostHog } from "posthog-js/react";
+
+export default function Test() {
+  const posthog = usePostHog();
+
+  return (
+    <main>
+      <h1>Hello from the bright side!</h1>
+      <p>Clicking this button will bring you happiness</p>
+      <button onClick={() => posthog.capture("main_button_clicked")}>
+        I want to be happy!
+      </button>
+    </main>
+  );
+}
+```
+
+Now run `npm run dev`. Go to each of our pages to see that they work: `http://localhost:3000/control` and `http://localhost:3000/test`. 
+
+Click the button on each page to capture a custom event in PostHog.
+
+![Events](../images/tutorials/redirect-testing/events.png)
+
+## Creating our A/B test
+
+Our A/B test will compare these two pages to see which drives more button clicks. To do this, we go to the [experiment tab](https://app.posthog.com/experiments) (our name for A/B tests) in PostHog and click "New experiment." Name your experiment and feature flag key (like `main-redirect`), set your experiment goal to `main_button_clicked`, and click "Save as draft."
+
+![A/B test set up](../images/tutorials/redirect-testing/test.png)
+
+Because we are working locally, you can click "Launch" immediately.
+
+## Setting up the redirect test middleware
+
+Next.js enables you to run [middleware](https://nextjs.org/docs/app/building-your-application/routing/middleware) that intercepts and modifies requests for your app. We use it for our redirect test.
```

Comment:
I think just a slighltly longer explanation is needed as to why we're using it. (My suggestion isnt perfect, but just to give an idea)
```suggestion
Next.js enables you to run [middleware](https://nextjs.org/docs/app/building-your-application/routing/middleware) that intercepts and modifies requests for your app. We use it for our redirect test so that we can redirect a user to the `control` or `test` page based on their experiment.
```